src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -952,6 +958,11 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
                 return _embeddedTypesManagerOpt.EmbedTypeIfNeedTo(namedTypeSymbol, fromImplements, syntaxNodeOpt, diagnostics);
             }
 
+            if (!InjectedSymbolsAreFrozen && namedTypeSymbol.Equals(_nonNullTypesAttribute))
+            {
+                NeedsNonNullTypesAttribute = true;
```

Comment:
>NeedsNonNullTypesAttribute = true; [](start = 16, length = 34)

Are we setting this property to true even if it is non an "injected" type?

Reply:
We do. But the injection into `GetTopLevelTypes` will skip the type if it's in source.

---
In reply to: [215102520](https://github.com/dotnet/roslyn/pull/29180#discussion_r215102520) [](ancestors = 215102520)